File: python/sparklabs/batch/sales/avg_product_price.py
from pyspark.sql import functions as F

from sparklabs.batch.sales import utils


def run() -> None:
    (
        utils.products.groupBy(F.col("category"))
        .agg(F.round(F.avg("price")).alias("avg_price"))
        .orderBy(F.desc("avg_price"))
        .show(truncate=False)
    )


if __name__ == "__main__":
    run()

File: scala/src/main/scala/io/devorbit/sparklabs/batch/sales/AvgProductPrice.scala
package io.devorbit.sparklabs.batch.sales

import org.apache.spark.sql.functions._

object AvgProductPrice extends App {
  Utils.products
    .groupBy(col("category"))
    .agg(round(avg("price")).as("avg_price"))
    .orderBy(desc("avg_price"))
    .show(false)

//  Thread.sleep(100000000)

}
